packages/stepper/src/Stepper.tsx

```diff
@@ -0,0 +1,521 @@
+
+  if (rangeStart > 0) {
+    stepRange.push({
+      'aria-label': 'Previous steps',
+      step: 'previous',
+      stepLabel: `+${rangeStart}`,
+      state: rangeStart > currentStep + 1 ? 'upcoming' : 'completed',
+      children: <EllipsisIcon />,
+    });
+  }
+
+  stepRange.push(
+    ...allSteps.slice(rangeStart, rangeEnd).map(step => ({ ...step })),
+  );
+
+  if (rangeEnd < allSteps.length) {
+    stepRange.push({
+      'aria-label': 'Next steps',
+      step: 'next',
+      stepLabel: `+${allSteps.length - rangeEnd}`,
+      state: 'upcoming',
+      children: <EllipsisIcon />,
+    });
+  }
+
+  return stepRange;
+}
+
+export default function Stepper({
+  children,
+  currentStep,
+  maxDisplayedSteps,
+}: StepperProps) {
+  const [visibleLayer, setVisibleLayer] = useState<
+    'previous' | 'next' | 'current'
+  >('current');
+
+  const allSteps: Array<StepItem> = (children instanceof Array
+    ? children
+    : [children]
+  ).map((child, index) => {
+    let state: StepCompletionState;
+
+    if (index < currentStep) {
+      state = 'completed';
+    } else if (index === currentStep) {
+      state = 'current';
+    } else {
+      state = 'upcoming';
+    }
+
+    return {
+      'aria-label': stepContentsAriaLabel(child.props.children),
+      'aria-current': currentStep === index ? 'step' : undefined,
+      step: index,
+      stepLabel: (index + 1).toString(),
+      state,
+      children: child,
+    };
+  });
+
+  const numSteps = allSteps.length;
+
+  const { rangeStart, rangeEnd } = computeRangeToDisplay({
+    currentStep,
+    numSteps,
+    maxDisplayedSteps,
+  });
+  const currentDisplayedSteps = getStepRange(allSteps, {
+    currentStep,
+    rangeStart,
+    rangeEnd,
+  });
+
+  const layerToLayerSteps: Record<string, Array<StepItem>> = {
+    current: currentDisplayedSteps,
+  };
+
+  if (rangeStart > 0) {
+    currentDisplayedSteps[0].onMouseOver = () => {
+      setVisibleLayer('previous');
+    };
+
+    layerToLayerSteps.previous = getStepRange(allSteps, {
+      currentStep,
+      ...computeRangeToDisplay({
+        currentStep: rangeStart - 1,
+        numSteps,
+        maxDisplayedSteps,
+      }),
+    });
+  }
+
+  if (rangeEnd < numSteps - 1) {
+    currentDisplayedSteps[
+      currentDisplayedSteps.length - 1
+    ].onMouseOver = () => {
+      setVisibleLayer('next');
+    };
+
+    layerToLayerSteps.next = getStepRange(allSteps, {
+      currentStep,
+      ...computeRangeToDisplay({
+        currentStep: rangeEnd,
+        numSteps,
+        maxDisplayedSteps,
+      }),
+    });
+  }
+
+  Object.entries(layerToLayerSteps).forEach(([layer, layerSteps]) => {
+    layerSteps.forEach((step, index) => {
+      if (index < layerSteps.length - 1) {
+        step.className = cx({
+          [slideInAnimationStyle]: visibleLayer === layer,
+          [slideOutAnimationStyle]: visibleLayer !== layer,
+        });
+      }
+    });
+  });
+
+  return (
+    <div
+      aria-label="breadcrumbs"
```

Comment:
Have been trying to find documentation to support this label, have you seen any? Open to it if it makes sense, but not sure that it does here